projects/objectnav_baselines/models/object_nav_models.py

```diff
@@ -125,71 +139,51 @@ def get_object_type_encoding(
             observations[self.goal_sensor_uuid].to(torch.int64)
         )
 
-    def forward(  # type:ignore
-        self,
-        observations: ObservationType,
-        memory: Memory,
-        prev_actions: torch.Tensor,
-        masks: torch.FloatTensor,
-    ) -> Tuple[ActorCriticOutput[DistributionType], Optional[Memory]]:
-        """Processes input batched observations to produce new actor and critic
-        values. Processes input batched observations (along with prior hidden
-        states, previous actions, and masks denoting which recurrent hidden
-        states should be masked) and returns an `ActorCriticOutput` object
-        containing the model's policy (distribution over actions) and
-        evaluation of the current state (value).
-
-        # Parameters
-        observations : Batched input observations.
-        memory : `Memory` containing the hidden states from initial timepoints.
-        prev_actions : Tensor of previous actions taken.
-        masks : Masks applied to hidden states. See `RNNStateEncoder`.
-        # Returns
-        Tuple of the `ActorCriticOutput` and recurrent hidden state.
-        """
+    def forward_encoder(self, observations: ObservationType) -> torch.FloatTensor:
         target_encoding = self.get_object_type_encoding(
             cast(Dict[str, torch.FloatTensor], observations)
         )
-        x = [target_encoding]
+        obs_embeds = [target_encoding]
 
         if not self.is_blind:
             perception_embed = self.visual_encoder(observations)
-            x = [perception_embed] + x
+            obs_embeds = [perception_embed] + obs_embeds
 
-        x_cat = torch.cat(x, dim=-1)  # type: ignore
-        x_out, rnn_hidden_states = self.state_encoder(
-            x_cat, memory.tensor("rnn"), masks
-        )
+        obs_embeds = torch.cat(obs_embeds, dim=-1)
+        return obs_embeds
 
-        return (
-            ActorCriticOutput(
-                distributions=self.actor(x_out), values=self.critic(x_out), extras={}
-            ),
-            memory.set_tensor("rnn", rnn_hidden_states),
-        )
 
-
-class ResnetTensorObjectNavActorCritic(ActorCriticModel[CategoricalDistr]):
+class ResnetTensorObjectNavActorCritic(VisualNavActorCritic):
```

Comment:
My guess is that, after these changes, our pretrained model (available [here](https://github.com/allenai/allenact/tree/main/projects/objectnav_baselines#robothor-objectnav-2021-challenge)) no longer works, correct? If so, can you add an override to the `load_state_dict` function that reformats the old weight format so that it can be loaded into the model? I.e.
```python
def load_state_dict(self, state_dict):
    if "some_key" in state_dict:
        # Old state dict format
        new_state_dict = somehow_relabel_state_dict(state_dict)
    else:
        #  Nothing to change
        new_state_dict = state_dict
    return super(self, ResnetTensorObjectNavActorCritic).load_state_dict(new_state_dict)
```

Reply:
It would be good to then run inference and double check that we can reproduce the old robothor objectnav numbers.

Reply:
Any luck with this?